client/app/components/account_config_main.coffee

```diff
@@ -0,0 +1,471 @@
+{
+    div, p, h3, h4, form, label, input, button, ul, li, a, span, i,
+    fieldset, legend
+} = React.DOM
+classer = React.addons.classSet
+
+AccountActionCreator = require '../actions/account_action_creator'
+AccountInput = require './account_config_input'
+
+RouterMixin = require '../mixins/router_mixin'
+LayoutActionCreator = require '../actions/layout_action_creator'
+{Form, FieldSet, FormButtons, FormDropdown} = require './basic_components'
+
+
+state = null
+
+module.exports = AccountConfigMain = React.createClass
+    displayName: 'AccountConfigMain'
+
+    mixins: [
+        RouterMixin
+        React.addons.LinkedStateMixin # two-way data binding
+    ]
+
+
+    # Do not update component if nothing has changed.
+    shouldComponentUpdate: (nextProps, nextState) ->
+        isNextState = _.isEqual nextState, @state
+        isNextProps = _.isEqual nextProps, @props
+        return not (isNextState and isNextProps)
+
+
+    _propsToState: (props) ->
+        state = props
+        return state
```

Comment:
All props are merged in to state, only props we want in state should be in the state to ease understanding of how the form works.